src/Core/Random/RandomPointSet.hpp

```diff
@@ -0,0 +1,152 @@
+#pragma once
+#include <Core/RaCore.hpp>
+
+#include <Core/Types.hpp>
+
+#include <random>
+
+namespace Ra {
+namespace Core {
+
+/** \brief Random point set utilities.
+ *
+ *  This namespace contains functions, classes and utilities for 1D, 2D and 3D random point sets
+ * management.
+ *
+ */
+namespace Random {
+
+Scalar constexpr sqrtNewtonRaphsonhelper( Scalar x, Scalar curr, Scalar prev ) {
+    return curr == prev ? curr : sqrtNewtonRaphsonhelper( x, 0.5_ra * ( curr + x / curr ), curr );
+}
+Scalar constexpr ct_sqrt( Scalar x ) {
+    return sqrtNewtonRaphsonhelper( x, x, 0_ra );
+}
+
```

Comment:
Are these suppose to be used directly by client code ? if not, maybe move to .cpp
if yes, rename ct_sqrt to something more explicit, and following camel case naming convention.

Reply:
These are used, right now, only by the private implementation of, e.g. FibonacciSequence to initialise `static constexpr Scalar phi` in the class definition, and should be a constexpr. So this need to be defined here. But, your right, this should be renamed.

Reply:
isn't it possible to move them as static constexpr private member function of FibonacciSequence ?

Reply:
yes for this PR. But I've some other code that uses these helpers. In this PR, Random utilities are only used for unittests. I've some other code, that will be future PR that uses these class and some of them use these helpers. I propose to add them in a namespace called "internal" in the "Random" namespace.

Reply:
I can also remove the commits about Random utilities from this PR

Reply:
Maybe can be easier to review, but I'm fine with your explaination also.

Reply:
So, as soon as you finish on these 3 files (and the associated unittest), I'll push a commit taking into account our discussions. I also can show you AFK what I did right now.